File: src/main/java/com/ject/studytrip/studylog/domain/factory/StudyLogFactory.java
package com.ject.studytrip.studylog.domain.factory;

import com.ject.studytrip.member.domain.model.Member;
import com.ject.studytrip.studylog.domain.model.StudyLog;
import com.ject.studytrip.trip.domain.model.DailyGoal;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class StudyLogFactory {
    public static StudyLog create(
            Member member, DailyGoal dailyGoal, String title, String content) {
        return StudyLog.of(member, dailyGoal, title, content);
    }
}

File: src/main/kotlin/com/ject/studytrip/studylog/domain/factory/StudyLogFactory.kt
package com.ject.studytrip.studylog.domain.factory

import com.ject.studytrip.member.domain.model.Member
import com.ject.studytrip.studylog.domain.model.StudyLog
import com.ject.studytrip.trip.domain.model.DailyGoal

object StudyLogFactory {
    fun create(
        member: Member,
        dailyGoal: DailyGoal,
        content: String,
    ): StudyLog = StudyLog.of(member, dailyGoal, content)
}
